src/integrations/watchers/indexable-home-page-watcher.php

```diff
@@ -108,6 +122,11 @@ public function check_option( $old_value, $new_value, $option ) {
 	 */
 	public function build_indexable() {
 		$indexable = $this->repository->find_for_home_page( false );
+
+		if ( $indexable === false && ! $this->indexable_helper->should_index_indexables() ) {
+			return;
+		}
+
 		$indexable = $this->builder->build_for_home_page( $indexable );
 
 		if ( $indexable ) {
```

Comment:
Underneath here we do $indexable->save() I am wondering if we should for super consistency sake also use the indexable helper here to call ->save_indexable() or was there a reason not to.